### DISCUSSION TITLE
Setting up tests

### DISCUSSION THREAD

**Discussion by mbobiosio** [oppia/oppia-android#4572] [Q&A (Installation)]
![image_2022-09-08_164753001](https://user-images.githubusercontent.com/31294354/189167212-ca03167f-bf01-4906-8a3f-440bc8e7beae.png)
I have followed the recommendations on [this discussion](https://github.com/oppia/oppia-android/discussions/4211) but it doesn't solve my issue.
Android Studio runs the build process successfully, shows "Test Passed" but no test in particular indicated. The app is not started on the device (emulator and mobile phone) during this process.

The expected output
![156389319-13ee965c-4c4f-40bc-82be-f6d32f0d9979](https://user-images.githubusercontent.com/31294354/189289712-6153d9d0-7785-4d27-a8e7-dfc5873028ce.png)

Android Studio version: Android Studio Arctic Fox | 2020.3.1
Java version: Java 8
OS: Ubuntu 22.04.1 LTS

**Comment by BenHenning** [comment]
If you run the test using the CLI in a local terminal, does it work? I think the command in your case would be:

```sh
./gradlew -Pandroid.testInstrumentationRunnerArguments.class=org.oppia.android.app.administratorcontrols.AdministratorControlsActivityTest :app:connectedAndroidTest
```

Also, have you checked the logcat logs on your device/emulator after starting the test? It may be the case that the app is misbehaving.

While https://github.com/google/dagger/issues/3223 was caused by something that couldn't break for us, it's noteworthy because the issue illustrated there might also affect us (specifically the fact that we use a custom runner which might not be playing nicely with the latest Android Studio or AGP version).

Edit: Sometimes this issue can come up if the app fails to start (or, in your case, install). You can force a clean state by manually uninstalling the app via the system's app package manager, or via ADB on the CLI using:

```sh
adb uninstall org.oppia.android
```

**Reply by mbobiosio** [reply]
CLI throws `com.android.ddmlib.InstallException` 

`com.android.builder.testing.api.DeviceException: com.android.ddmlib.InstallException: INSTALL_FAILED_UPDATE_INCOMPATIBLE: Package org.oppia.android.test signatures do not match previously installed version; ignoring!`

I have used 'clear app data - uninstall' and rerun test but still the same outcome. This happens both for emulator and mobile.

**Reply by BenHenning** [reply]
Try fully uninstalling the app either through the device's app package manager or via:

```sh
adb uninstall org.oppia.android
```

I am almost certain this is the issue that you're running into.

**Reply by mbobiosio** [reply]
This doesn't solve my issue unfortunately

**Reply by BenHenning** [reply]
@mbobiosio could you include a bit more details, e.g.: are you still experience the exact same stack trace? If not, what new error are you seeing (as it may be an unrelated problem to the original one asked in this discussion).

**Comment by MohitGupta121** [comment]
@mbobiosio I think you have to downgrade your Android Studio to Bumblebee to run the tests on that or check this [disscusion](https://github.com/oppia/oppia-android/discussions/4432) here you able to find the solution.

**Reply by MohitGupta121** [reply]
@mbobiosio Is still you getting that error or is fixed?
If it fixed by this answer please mark this as answer.
Or if you find any other way that work for you please share and mark that as answer.
So that this helps other in community.

**Reply by MohitGupta121** [reply]
Hey @mbobiosio if your issue is resolved please mark this discussion is resolved (marking as answered that comment) so that it's helpful to others in the community.
And if you need help then also feel free to ask.
